packages/dds/tree/src/feature-libraries/chunked-forest/basicChunk.ts

```diff
@@ -107,6 +107,7 @@ export class BasicChunkCursor extends SynchronousCursor implements ChunkedCursor
 		protected index: number,
 		protected indexOfChunk: number,
 		protected indexWithinChunk: number,
+		protected nestedCursor: ChunkedCursor | undefined,
```

Comment:
Can you add a doc for this? In particular, what the expected behavior of having a nested cursor is. 

In general, this pattern doesn't seem very intuitive to me. If this cursor does have children that are BasicChunks but also gets a nested cursor, are the BasicChunks just completely ignored? Does having a nested cursor just completely override the rest of the cursor's normal functionality?